### PR TITLE
Add indium-update-script-source-hook

### PR DESCRIPTION
**Pull request by TatriX**
Usage:
```lisp
(add-hook 'indium-update-script-source-hook
	  (lambda (url)
	    (indium-eval (format "window.dispatchEvent(new CustomEvent('patch', {detail: {url: '%s'}}))"
				 url))))
```
and
```js
window.addEventListener("patch", (event) => {
    console.log("Patched @ " + new Date().toTimeString().substring(0, 8), event.detail);
    // rerender, etc
});

```

### CODE REVIEW

**Comment by NicolasPetton** [comment]
Looks good, thanks!

Could you also add documentation for it?

**Comment by TatriX** [comment]
To the texi file?

**Comment by NicolasPetton** [comment]
No, it should be placed in `sphinx-doc/code-evaluation.rst`.
Then from the `sphinx-doc` directory, you can run `make  texinfo` (or `M-x compile RET make texinfo RET` to update the texinfo manual.

File: sphinx-doc/code-evaluation.rst [hunk 1]
```diff
@@ -68,3 +68,17 @@ To enable live updates, make sure Indium is set up to use local files (see
...
   script JavaScript source after each buffer save.
+
+You can setup a hook for the script update. For example ::
+
+  (add-hook 'indium-update-script-source-hook
+	  (lambda (url)
+	    (indium-eval (format "window.dispatchEvent(new CustomEvent('patch', {detail: {url: '%s'}}))"
+				 url))))
+
+Then you can use it in your app for developmenm purposes ::
```
Comment (NicolasPetton) [inline sphinx-doc/code-evaluation.rst]: there's a typo here :)

File: sphinx-doc/code-evaluation.rst [hunk 2]
```diff
@@ -68,3 +68,17 @@ To enable live updates, make sure Indium is set up to use local files (see
   current buffer.
 - ``(setq indium-update-script-on-save t)``: Automatically update the runtime
   script JavaScript source after each buffer save.
+
+You can setup a hook for the script update. For example ::
```
Comment (NicolasPetton) [inline sphinx-doc/code-evaluation.rst]: What about rephrasing this sentence like this: `You can setup a hook to be run after each script update. For example ::`

**Comment by NicolasPetton** [comment]
Squash your commits together and we're good to go!

**Pull request merged**